AppSyncSubscriptionClient/AppSyncSubscriptionClient/ConnectionProvider/AppsyncRealtimeConnection/RealtimeConnectionProvider+Websocket.swift

```diff
@@ -10,11 +10,11 @@ import Foundation
 extension RealtimeConnectionProvider: AppSyncWebsocketDelegate {
 
     public func websocketDidConnect(provider: AppSyncWebsocketProvider) {
+        lastKeepAliveTime = DispatchTime.now()
```

Comment:
I think that keep-alive messages arrive on `WebsocketDidReceiveMessage`, so I'm a bit surprised to see this in the websocketDidConnect method.

Reply:
`lastKeepAliveTime` is technically "last time we got any data back" which includes graphql responses, connection messages, errors, etc. Once the keepAlive process checks that current time is greater than ("lastKeepAliveTime" + 5 minutes) then it is determined to be stale and sets the internal connection provider state to .notConnected and best effort call to `websocket.disconnect()`. Since introducing this change to check count of subscriptions on the socket, we hold onto the connection provider instance while keeping the websocket disconnected. 

Looking at this further, originally I added this so to account for the case where a websocket connection is established again at a later time. but in fact we don't need this since the when we process the response from the websocket service, we will also update it then. so it's sort of redundant to update it twice, once at connect and once at websocketDidReceivedData. I will remove this check here